content/what-is-roles-based-access-control-vs-abac/index.md

```diff
@@ -0,0 +1,66 @@
+
+While ABAC gives finer granularity over access to different resources, RBAC wins out on the ease of implementation. The initial setup cost of RBAC is magnitudes lower than ABAC.
+
+With RBAC, companies have a straightforward method for grouping users with similar access needs and then assigning permissions to those groups.
+
+With ABAC, companies must first define variables and configure different attribute-based rules, which can be a cumbersome process.
+
+![RBAC vs ABAC](./rbac-abac.png)
+
+
+## Advantages of RBAC
+
+Easy to understand: Because of its intuitive structure, the underlying business logic with RBAC is simple to communicate and understand – even with dozens of different roles.
+
+- Extendibility: As you ship new features or change the application structure, adding or modifying roles can easily extend access to new resources for users that need them.
+- Improving compliance: Using RBAC forces developers to think about and organize application permissions and access control. This information can then be used by compliance officers during audits.
+- Decreased risk of data breaches / leakage: It becomes easy for developers to implement application security best practices around access control for their APIs, greatly reducing the chances of breaches.
+
+## Disadvantages of RBAC
+- Difficult to make exceptions: It can be complex to make exceptions to how a role works. In our example above, if we want to add a rule that users with `regular-user` role cannot edit their own post if they have already made ten edits, it will have to be added in the API logic as an exception. There is no way for the roles / permissions system to express it easily. This causes issues since we have to encode this rule in all places where we are checking for the `edit:self` permission.
+- Role explosion: Because the only way to add granularity to a RBAC system is by creating a new role, we can end up with hundreds of different roles that are impossible to manage.
+- Can cause conflicts in permissions: There could be situations in which a user is assigned two roles that have conflicting information. In our example, if a user is assigned the `admin` role and the `regular-user` role, they have `edit:self` and `edit:all` permissions to edit blog posts. Which one should take precedence? The precedence logic can be coded in the APIs, but this opens the possibility for errors.  
+  
+```ts
+if (permissions.contains("edit:self")) {
+	// only allow editing if the blog post belongs to the current user 
+} 
+else if (permissions.contains("edit:all")) {
+	// allow editing 
+}  
+```  
+
+Even thought our user has both the `admin` and `regular-user` roles, they will not be able to edit all the blogs because the `edit:self` statement is executed first while the `edit:all` rule get skipped in the following else-if statement.
```

Comment:
```suggestion
Even though our user has both the `admin` and `regular-user` roles, they will not be able to edit all the blogs because the `edit:self` statement is executed first while the `edit:all` rule get skipped in the following else-if statement.
```